hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java

```diff
@@ -597,7 +592,7 @@ public void addCallSize(final long diff) {
    * @param addr InetAddress of incoming connection
    * @throws AuthorizationException when the client isn't authorized to talk the protocol
    */
-  public void authorize(UserGroupInformation user, ConnectionHeader connection,
+  public synchronized void authorize(UserGroupInformation user, ConnectionHeader connection,
```

Comment:
Add synchronized to this method,  the " synchronized (authManager) { } " in the method can removed?

Reply:
Oh yes, should do. Will fix in next commit.